src/views/todos.js

```diff
@@ -18,16 +18,19 @@ function vrenderHeader(todosData) {
 }
 
 function vrenderMainSection(todosData) {
-  let allCompleted = todosData.list.reduce((x, y) => x && y.completed, true);
+  let list = todosData.get('list');
+  let allCompleted = list.every(x => x.get('completed'));
+
   return h('section#main', {
-    style: {'display': todosData.list.length ? '' : 'none'}
+    style: {'display': list.size ? '' : 'none'}
   }, [
     h('input#toggle-all', {
       type: 'checkbox',
       checked: allCompleted
     }),
-    h('ul#todo-list', todosData.list
-      .filter(todosData.filterFn)
+    h('ul#todo-list', list
+      .filter(todosData.get('filterFn'))
+      .toJS()
```

Comment:
I don't know if it is better to case to JS before the map or after.
Doesn't matter that much I guess. My thought was that it might be more performant this way.